tests/5.0/tasks/test_task_depend_iterator.cpp

```diff
@@ -0,0 +1,61 @@
+//===-- test_task_depend_iterator.cpp - test of the iterator clause inside the task depend clause ----===//
+// 
+// OpenMP API Version 5.0 Nov 2020
+// 
+// This file is a test for the iterator modifier when used with the task depend
+// clause. This modifier should create an iterator that expand into multiple values
+// inside the clause they appear. In this particular test case the iterator expands into
+// several values creating several dependencies at the same time.
+//
+//===-------------------------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <algorithm>
+#include <chrono>
+#include <cstdlib>
+#include <iostream>
+#include <thread>
+#include <vector>
+#include "ompvv.h"
+
+int test_task_depend_iterator() {
+  int ptr[] = {0, 4, 5, 6, 7, 8, 9, 10, 11}, cols[] = {1, 2, 3, 4, 5, 5, 6, 6, 7, 7, 8};
+  std::vector<int> threadOrder, threadAssignment;
+  bool threadOrderError=false, threadAssignmentWarning=false;
+#pragma omp parallel num_threads(4)
+  {
+#pragma omp single
+    {
+      for (int i = 0; i < 8; ++i) {
+        int pos = ptr[i], size = ptr[i+1] - ptr[i];
+#pragma omp task depend(iterator(it = 0 : size), in : ptr[cols[pos+it]]) depend(out : ptr[i])
+        {
+#pragma omp critical
+          {
+            threadOrder.push_back(i);
+            threadAssignment.push_back(omp_get_thread_num());
+          } // end critical section
+          std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        } // end task depend
+      }
+    } // end single
+  } // end parallel
+  std::sort(threadOrder.begin()+1,threadOrder.begin()+5);
+  std::sort(threadOrder.begin()+5,threadOrder.begin()+7);
+  std::sort(threadAssignment.begin()+1,threadAssignment.begin()+5);
+  for(int i = 0; i < 8; ++i)
+    threadOrderError = (threadOrder[i] != i) || threadOrderError;
+  for(int i = 0; i < 4; ++i)
```

Comment:
There is a chance the test will run with less than threads since the specification does not enforce the number of threads to be equal to `num_threads()`, just never more than `num_threads()`. So I suggest you use `omp_get_num_threads()` to get the actual number of threads and use it for this for loop upper limit.

Reply:
I know, that is why I set it as a warning because if the depend implementation is incorrect it may use only two threads even if launched with 4 threads. However since there is no specification guarantee may be remove that portion of the test?

Reply:
I think it is fine to have this test, it's just a matter of adapting it to the actual number of threads during execution, and that should be it.

Reply:
What I mean is that if the dependencies are not enforced correctly by the implementation then even when there are 4 active threads launched by OMP it may report only using two threads. I think since there are no specification guarantees it may be better to leave that part out.